### PR TITLE
BVT-VERIFY-UUID-FSTAB: fix for SLES12

### PR DESCRIPTION
**Pull request by v-sijliu**
Fix test cases BVT-VERIFY-UUID-FSTAB against SLES12.3 and SLES12.4

### CODE REVIEW

File: Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py [hunk 1]
```diff
@@ -13,10 +13,14 @@ def RunTest():
     output = output.lower()
     filter_condition_dmesg = r'.*root=UUID=(.*?) .*'
     filter_condition_fstab = r'.*UUID=(.*?)[ \t]+\/[ \t]+'
-    if (DetectDistro()[0] == 'opensuse' or DetectDistro()[0] == 'SUSE' or (DetectDistro()[0] == 'sles' and DetectDistro()[1] != '15')):
+    if (DetectDistro()[0] == 'opensuse' or DetectDistro()[0] == 'SUSE' or ((DetectDistro()[0] == 'sles' and DetectDistro()[1] != '15') and (DetectDistro()[0] == 'sles' and DetectDistro()[1] != '12.4'))):
         filter_condition_dmesg = r'.*root=/dev/disk/by-uuid/(.*?) .*'
         filter_condition_fstab = r'.*/dev/disk/by-uuid/(.*?)[ \t]+\/[ \t]+'
 
+    if (DetectDistro()[0] == 'opensuse' or DetectDistro()[0] == 'SUSE' or (DetectDistro()[0] == 'sles' and DetectDistro()[1] == '12.3')):
```
Comment (lizzha) [inline Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: don't quite get the logic here, is it for sles 12.3 only? you're overwriting for openSUSE and SUSE as well

File: Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py [hunk 2]
```diff
@@ -13,10 +13,14 @@ def RunTest():
     output = output.lower()
     filter_condition_dmesg = r'.*root=UUID=(.*?) .*'
     filter_condition_fstab = r'.*UUID=(.*?)[ \t]+\/[ \t]+'
-    if (DetectDistro()[0] == 'opensuse' or DetectDistro()[0] == 'SUSE' or (DetectDistro()[0] == 'sles' and DetectDistro()[1] != '15')):
+    if (DetectDistro()[0] == 'opensuse' or DetectDistro()[0] == 'SUSE' or ((DetectDistro()[0] == 'sles' and DetectDistro()[1] != '15') and (DetectDistro()[0] == 'sles' and DetectDistro()[1] != '12.4'))):
```
Comment (lizzha) [inline Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: which is the supported sles version here?

**Comment by v-sijliu** [comment]
BVT-VERIFY-UUID-FSTAB  | SLES   12.4 | PASS       
                                          | SLES   12.3 | PASS

[LISAv2 Test Results Summary]
Test Run On           : 02/25/2019 07:39:18
ARM Image Under Test  : SUSE : SLES : 12-SP3 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:14

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 3.21 

[LISAv2 Test Results Summary]
Test Run On           : 02/25/2019 07:39:23
ARM Image Under Test  : SUSE : SLES : 12-SP4 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:11

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 3.25

File: Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py (hunk 1 above)
Comment (LiliDeng) [reply Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: DetectDistro()[0] == 'sles' and DetectDistro()[1] == '12.3' should be enough? @v-sijliu

File: Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py [hunk 3]
```diff
@@ -13,10 +13,14 @@ def RunTest():
     output = output.lower()
     filter_condition_dmesg = r'.*root=UUID=(.*?) .*'
     filter_condition_fstab = r'.*UUID=(.*?)[ \t]+\/[ \t]+'
-    if (DetectDistro()[0] == 'opensuse' or DetectDistro()[0] == 'SUSE' or (DetectDistro()[0] == 'sles' and DetectDistro()[1] != '15')):
+    if (DetectDistro()[0] == 'opensuse' or DetectDistro()[0] == 'SUSE'):
         filter_condition_dmesg = r'.*root=/dev/disk/by-uuid/(.*?) .*'
         filter_condition_fstab = r'.*/dev/disk/by-uuid/(.*?)[ \t]+\/[ \t]+'
 
+    if (DetectDistro()[0] == 'sles' and DetectDistro()[1] == '12.3'):
```
Comment (lizzha) [inline Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: how about other sles versions?

Comment (v-sijliu) [reply Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: There are 4 skus for sles, sles 11 sp4, sles 12 sp3, sles 12 sp4 and sles 15, should we test sles 11 sp4?

Comment (lizzha) [reply Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: if it's in azure marketplace, we should also include it

Comment (chvalean) [reply Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: IMO we should not do per release checks.
The automation must be smart enough to detect 2 patterns - differences between releases.
I know this is just an addition, so not sure if a refactor of this test is wanted...

Comment (lizzha) [reply Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: agree that it would be preferable to detect per pattern matches, instead of checking distros

**Comment by LiliDeng** [comment]
Part of work #756

File: Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py [hunk 4]
```diff
@@ -11,11 +11,8 @@ def RunTest():
     dmsg_dev_count = 0
     output = JustRun("dmesg")
     output = output.lower()
-    filter_condition_dmesg = r'.*root=UUID=(.*?) .*'
-    filter_condition_fstab = r'.*UUID=(.*?)[ \t]+\/[ \t]+'
-    if (DetectDistro()[0] == 'opensuse' or DetectDistro()[0] == 'SUSE' or (DetectDistro()[0] == 'sles' and DetectDistro()[1] != '15')):
-        filter_condition_dmesg = r'.*root=/dev/disk/by-uuid/(.*?) .*'
-        filter_condition_fstab = r'.*/dev/disk/by-uuid/(.*?)[ \t]+\/[ \t]+'
+    filter_condition_dmesg = r'.*root=UUID=(.*?) .*|.*uuid-(.*?)\..*|.*root=/dev/disk/by-uuid/(.*?) .*'
```
Comment (ader1990) [inline Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: can you create a note and explain what these regexes do? these regexes got very completed and I think it s time for a small explanation on what they do.

Comment (LiliDeng) [reply Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: sure, multiple if will make logic clear but not smart, simple code looks better but hard to understand.

File: Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py [hunk 5]
```diff
@@ -8,36 +8,52 @@
...
     fstab_dev_count = 0
     fstab_dev_count = output = JustRun("cat /etc/fstab")
     fstab_dev_count = output.count('/dev/sd')
 
     outputlist = re.split("\n", output)
     for line in outputlist:
-        matchObj = re.match(filter_condition_fstab, line, re.M|re.I)
-        #matchObj = re.match( r'.*UUID=(.*?)[ \t]*/ .*', line, re.M|re.I)
+        matchObj = re.search(filter_condition_fstab, line, re.IGNORECASE)
+        #matchObj = re.search( r'uuid(/|=)(\S+)[ \t]+\/(.*?)[ \t]+(.*?)[ \t]', line, re.IGNORECASE)
```
Comment (lizzha) [inline Testscripts/Linux/BVT-VERIFY-UUID-FSTAB.py]: if it's not needed, the commented line can be removed

**Comment by LiliDeng** [comment]
Done for testing

```
[LISAv2 Test Results Summary]
Test Run On           : 03/01/2019 09:51:40
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:11

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.59 

[LISAv2 Test Results Summary]
Test Run On           : 03/01/2019 10:03:24
ARM Image Under Test  : CoreOS : CoreOS : Stable : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:11

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.65 


[LISAv2 Test Results Summary]
Test Run On           : 03/01/2019 10:15:00
ARM Image Under Test  : SUSE : SLES : 15 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:16

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.64 

[LISAv2 Test Results Summary]
Test Run On           : 03/01/2019 10:31:08
ARM Image Under Test  : SUSE : SLES : 12-SP4 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:12

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.61 

[LISAv2 Test Results Summary]
Test Run On           : 03/01/2019 10:43:16
ARM Image Under Test  : SUSE : SLES : 12-SP3 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:13

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.59 
```

**Comment by lizzha** [comment]
@LiliDeng have you covered test on centos/rhel?

**Comment by LiliDeng** [comment]
Failed against redhat75, need update code again @v-sijliu @lizzha

**Comment by LiliDeng** [comment]
Pass against ubuntu, redhat, centos, sles, clearos, coreos, Debian, oracle.
```
[LISAv2 Test Results Summary]
Test Run On           : 03/04/2019 02:55:39
ARM Image Under Test  : RedHat : RHEL : 7.5 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:12

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.83 

[LISAv2 Test Results Summary]
Test Run On           : 03/04/2019 03:08:36
ARM Image Under Test  : credativ : Debian : 9-backports : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:12

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.62 

[LISAv2 Test Results Summary]
Test Run On           : 03/04/2019 03:32:41
ARM Image Under Test  : SUSE : SLES : 15 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:17

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.66 [LISAv2 Test Results Summary]

Test Run On           : 03/04/2019 03:49:46
ARM Image Under Test  : SUSE : SLES : 12-SP4 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:12

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.61 

[LISAv2 Test Results Summary]
Test Run On           : 03/04/2019 04:21:04
ARM Image Under Test  : SUSE : SLES : 12-SP3 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:14

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.64 

[LISAv2 Test Results Summary]
Test Run On           : 03/04/2019 04:36:00
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:10

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                  2.6 

[LISAv2 Test Results Summary]
Test Run On           : 03/04/2019 04:47:13
ARM Image Under Test  : OpenLogic : CentOS : 7.5 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:10

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.74 


[LISAv2 Test Results Summary]
Test Run On           : 03/04/2019 04:57:48
ARM Image Under Test  : Oracle : Oracle-Linux : 7.6 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:12

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.71 


[LISAv2 Test Results Summary]
Test Run On           : 03/04/2019 05:10:34
ARM Image Under Test  : CoreOS : CoreOS : Stable : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:11

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 2.71 


[LISAv2 Test Results Summary]
Test Run On           : 03/04/2019 05:22:32
ARM Image Under Test  : clear-linux-project : clear-linux-os : basic : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:13

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-UUID-FSTAB                                                             PASS                 3.48 
```

**Pull request merged**